### PR TITLE
Overlord

### PR DESCRIPTION
**Pull request by arrubin**

### CODE REVIEW

**Comment by arrubin** [comment]
All of Hound's complaints are about long descriptions in my RSpec tests. I am open to advice on how to shorten these.

File: overlord/features/step_definitions/overlord_steps.rb [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+When(/^I enter the deactivation code$/) do
+  fill_in "deactivation_code", with: "0000"
+end
+
+When(/^I submit the setup information$/) do
+  click_button "Save"
+end
+
+Given(/^I have setup the bomb$/) do
```
Comment (csain) [inline overlord/features/step_definitions/overlord_steps.rb]: These are the same codes used as defaults. You may also want to test with the custom codes to differentiate that the custome codes feature works properly

File: overlord/lib/overlord/bomb.rb [hunk 2]
```diff
@@ -0,0 +1,60 @@
+module Overlord
+  class Bomb
+    attr_reader :status, :errors
+
+    def initialize
+      @status = "Setup"
+      @errors = []
+      @failed_deactivations = 0
```
Comment (csain) [inline overlord/lib/overlord/bomb.rb]: Can you re-activate the bomb once it has been deactivated? If so, you may want this setter in `activate`

File: overlord/lib/overlord/bomb.rb [hunk 3]
```diff
@@ -0,0 +1,60 @@
+module Overlord
+  class Bomb
+    attr_reader :status, :errors
+
+    def initialize
+      @status = "Setup"
+      @errors = []
```
Comment (csain) [inline overlord/lib/overlord/bomb.rb]: `@errors = []` occurs multiple times. You might want to DRY-it-out with a method `reset_errors`

File: overlord/spec/overlord/bomb_spec.rb [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+        expect(bomb.status).to eq("Setup")
+      end
+    end
+
+    describe "#setup" do
+      it "sets status to 'Ready' when called with valid codes" do
+        bomb.setup("1234", "0000")
+        expect(bomb.status).to eq("Ready")
+      end
+      it "sets status to 'Ready' when called with blank codes" do
```
Comment (csain) [inline overlord/spec/overlord/bomb_spec.rb]: What happens with spaces? `bomb.setup("  ", "  ")`

**Comment by csain** [comment]
This is good, very clean and easy to understand. Just had a few minor comments to review

**Comment by djkotowski** [comment]
Looks good. Take a look at the comments, but no need for resubmission.

**Pull request closed without merging**